src/net_socket.c

```diff
@@ -704,7 +701,6 @@ void handle_new_meta_connection(event_loop_t *loop, void *data, int flags) {
 
 	c = new_connection();
 	c->name = xstrdup("<unknown>");
-	c->outcompression = mesh->self->connection->outcompression;
```

Comment:
What does this `outcompression` and `incompression` in node handle and connection handle respectively deal with?

Reply:
The `incompression` and `outcompression` variables are actually not used anywhere at the moment (I will remove them in a separate commit), but I had to remove this line because `mesh->self->connection` is now always `NULL` in this commit.